modules/samples/firealarm/component/plugin/src/main/java/org/homeautomation/firealarm/plugin/impl/dao/impl/DeviceTypeDAOImpl.java

```diff
@@ -0,0 +1,239 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.homeautomation.firealarm.plugin.impl.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.homeautomation.firealarm.plugin.constants.DeviceTypeConstants;
+import org.homeautomation.firealarm.plugin.exception.FirealarmPluginException;
+import org.homeautomation.firealarm.plugin.impl.dao.DeviceTypeDAO;
+import org.homeautomation.firealarm.plugin.impl.dao.util.DeviceTypeUtils;
+import org.wso2.carbon.device.mgt.common.Device;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Device Dao for Firealarm Devices.
+ */
+public class DeviceTypeDAOImpl {
+
+    private static final Log log = LogFactory.getLog(DeviceTypeDAOImpl.class);
+
+    public Device getDevice(String deviceId) throws FirealarmPluginException {
+        Connection conn;
+        PreparedStatement stmt = null;
+        Device connectedCupDevice = null;
+        ResultSet resultSet = null;
+        try {
+            conn = DeviceTypeDAO.getConnection();
+            String selectDBQuery =
+                    "SELECT firealarm_DEVICE_ID, DEVICE_NAME, ACCESS_TOKEN, REFRESH_TOKEN" +
+                    " FROM firealarm_DEVICE WHERE firealarm_DEVICE_ID = ?";
+            stmt = conn.prepareStatement(selectDBQuery);
+            stmt.setString(1, deviceId);
+            resultSet = stmt.executeQuery();
+
+            if (resultSet.next()) {
+                connectedCupDevice = new Device();
+                connectedCupDevice.setName(resultSet.getString(
+                        DeviceTypeConstants.DEVICE_PLUGIN_DEVICE_NAME));
+                List<Device.Property> propertyList = new ArrayList<>();
+                propertyList.add(DeviceTypeUtils.getProperty(
+                        DeviceTypeConstants.DEVICE_PLUGIN_PROPERTY_ACCESS_TOKEN,
+                        resultSet.getString("ACCESS_TOKEN")));
+                propertyList.add(DeviceTypeUtils.getProperty(
+                        DeviceTypeConstants.DEVICE_PLUGIN_PROPERTY_REFRESH_TOKEN,
+                        resultSet.getString("REFRESH_TOKEN")));
+                connectedCupDevice.setProperties(propertyList);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Firealarm service " + deviceId + " data has been fetched from" +
+                              "Firealarm database.");
+                }
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while fetching Firealarm device : '" + deviceId + "'";
+            log.error(msg, e);
+            throw new FirealarmPluginException(msg, e);
+        } finally {
+            DeviceTypeUtils.cleanupResources(stmt, resultSet);
+            DeviceTypeDAO.closeConnection();
```

Comment:
normally a .closeConnection() should occur at the business logic layer, not at the DAO layer.